scripts/clan_resources/herb/herb_supply.py

```diff
@@ -0,0 +1,728 @@
+            weight = [1, 2, 3]
+        else:
+            weight = [5, 2, 1]
+
+        # the amount of herb types the med has found
+        amount_of_herbs = choices(population=[1, 2, 3], weights=weight, k=1)[0] + amount_modifier
+        if general_amount_bonus:
+            amount_of_herbs *= 2
+
+        # adding herb quantity bonus
+        if specific_quantity_bonus:
+            quantity_modifier *= specific_quantity_bonus
+
+        # now we find what herbs have actually been found and their quantity
+        for herb in herb_list:
+            if amount_of_herbs == 0:
+                break
+
+            # rarity is set to 0 if the herb can't be found in the current season
+            if not self.herb[herb].get_rarity(game.clan.biome, game.clan.current_season):
+                continue
+
+            # chance to find a herb is based on it's rarity
+            if randint(1, self.herb[herb].get_rarity(game.clan.biome, game.clan.current_season)) == 1:
+                found_herbs[herb] = choices(population=[1, 2, 3], weights=weight, k=1)[0] * quantity_modifier
+                amount_of_herbs -= 1
+
+        list_of_herb_strs = []
+
+        if found_herbs:
+            for herb, count in found_herbs.items():
+                # add it to the collection
+                self.add_herb(herb, count)
+
+                # figure out how grammar needs to work in log
+                if count > 1:
+                    list_of_herb_strs.append(f"{count} {self.herb[herb].plural_display}")
+                else:
+                    list_of_herb_strs.append(f"{count} {self.herb[herb].singular_display}")
+
+        return list_of_herb_strs, found_herbs
+
+    def _add_collection_to_storage(self, med_cats):
+        """
+        Adds herbs in self.collected to self.storage and clears self.collected dict. If Clan has a CAMP skill med,
+        this is where an expiration perk is added.
+        """
+        # add empty "clump" to all uncollected herb stores
+        for herb in [x for x in self.base_herb_list if x not in self.collected and x in self.storage]:
+            self.storage.get(herb, []).insert(0, 0)
+
+        for med in med_cats:
+            # check if any meds can use a skill to store herbs better (aka, reduce time to expire)
+            # we base this modifier on their path points,
+            # this means there's skill variation even within cats with matching tiers
+            if med.skills.primary.path == SkillPath.CAMP:
+                modifier = med.skills.primary.points
+            elif med.skills.secondary and med.skills.secondary.path == SkillPath.CAMP:
+                modifier = med.skills.secondary.points
+            else:
+                continue
+
+            better_storage = []
+            for herb, count in self.collected.items():
+                if not self.storage.get(herb, []):
+                    # herbs can't be stored better if there isn't an existing store of that herb
+                    break
```

Comment:
```suggestion
                    continue
```
Unless it's intended that this should end the moment we find an empty storage